### PR TITLE
Midround infiltrators for ghosts/observers

### PR DESCRIPTION
**Pull request by SmoSmoSmoSmok**
## About The Pull Request

 MIDround infilt for observers i mightve done dumb shit when trying to add this i CANT wait to get bullied on the review s

## Why It's Good For The Game

  ahahahahahahahahhahahahhahahahhahahahhahahahhahahahhahahahhahahahhahahahhahahahhahahahhahahahhahahahhahahahhahahahhahahahhahahahhahahahhahahahhahahahhahahah

## Changelog

:cl:
add: Added new mechanics or gameplay changes
add: Added more things
expansion: Expands content of an existing feature
del: Removed old things
qol: made something easier to use
balance: rebalanced something
fix: fixed a few things
soundadd: added a new sound thingy
sounddel: removed an old sound thingy
imageadd: added some icons and images
imagedel: deleted some icons and images
spellcheck: fixed a few typos
code: changed some code
refactor: refactored some code
config: changed some config setting
admin: messed with admin stuff
server: something server ops should know
/:cl:

### CODE REVIEW

File: fulp_modules/features/antagonists/infiltrator/infil_ruleset.dm [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		if(!isturf(carp_spawn.loc))
+			stack_trace("Carp spawn found not on a turf: [carp_spawn.type] on [isnull(carp_spawn.loc) ? "null" : carp_spawn.loc.type]")
+			continue
+		spawn_locs += carp_spawn.loc
+	if(!spawn_locs.len)
+		message_admins("No valid spawn locations found, aborting...")
+		return MAP_ERROR
+	return ..()
+
+/datum/round_event_control/infiltrator
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrator/infil_ruleset.dm:30]: why round event too?

File: fulp_modules/features/antagonists/infiltrator/infil_ruleset.dm [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	infiltrator.mind.add_antag_datum(/datum/antagonist/traitor/infiltrator)
+	spawned_mobs += infiltrator
+	message_admins("[ADMIN_LOOKUPFLW(infiltrator)] has been made into a infiltrator by an event.")
+	log_game("[key_name(infiltrator)] was spawned as a infiltrator by an event.")
+
+	return SUCCESSFUL_SPAWN
+
+/proc/create_infiltrator(spawn_loc)
+	var/mob/living/carbon/human/infiltrator = new(spawn_loc)
+	infiltrator.randomize_human_appearance(~(RANDOMIZE_NAME|RANDOMIZE_SPECIES))
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrator/infil_ruleset.dm:73]: why keep the name?

File: fulp_modules/features/antagonists/infiltrator/infiltrators.dm [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	if(give_equipment)
+		equip_infiltrator(owner.current)
+	owner.current.mind.set_assigned_role(SSjob.GetJobType(/datum/job/infiltrator))
+	owner.current.mind.special_role = ROLE_INFILTRATOR
+	infil_flavor = strings("infiltrator_flavor.json", "infiltrator", "fulp_modules/strings/infils")
+	..()
+
+/datum/antagonist/traitor/infiltrator/greet()
+	. = ..()
+	owner.announce_objectives()
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrator/infiltrators.dm:32]: is this needed?

File: fulp_modules/features/antagonists/infiltrator/infiltrators.dm [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+/datum/antagonist/traitor/infiltrator/ui_static_data(mob/user)
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrator/infiltrators.dm:34]: What's the difference between this and traitor that makes it worth giving its own ui_static_data?

File: fulp_modules/features/antagonists/infiltrator/infiltrators.dm [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	uniform = /obj/item/clothing/under/chameleon
+	suit = /obj/item/clothing/suit/space/syndicate
+	glasses = /obj/item/clothing/glasses/night
+	mask = /obj/item/clothing/mask/gas/syndicate
+	head = /obj/item/clothing/head/helmet/space/syndicate
+	ears = /obj/item/radio/headset
+	back = /obj/item/storage/backpack/old
+	backpack_contents = list(/obj/item/storage/box/survival/syndie=1,\
+		/obj/item/knife/combat/survival,
+	)
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrator/infiltrators.dm:65]:
```suggestion
	backpack_contents = list(
		/obj/item/storage/box/survival/syndie = 1,
		/obj/item/knife/combat/survival = 1,
	)
```

File: fulp_modules/features/antagonists/infiltrator/infiltrators.dm [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+	id = /obj/item/card/id/advanced/chameleon
+	l_hand = /obj/item/tank/jetpack
+	shoes = /obj/item/clothing/shoes/jackboots
+	gloves = /obj/item/clothing/gloves/combat
+	r_pocket = /obj/item/tank/internals/emergency_oxygen
+	internals_slot = ITEM_SLOT_RPOCKET
+	belt = /obj/item/crowbar/power/syndicate
+	implants = list(/obj/item/implant/explosive)
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrator/infiltrators.dm:73]: Why do they get a microbomb as a non-nukie? Maybe in their uplink but...

File: tgui/packages/tgui/interfaces/PreferencesMenu/antagonists/antagonists/ghostinfiltrator.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import { Antagonist, Category } from "../base";
```
Comment (JohnFulpWillard) [inline tgui/packages/tgui/interfaces/PreferencesMenu/antagonists/antagonists/ghostinfiltrator.ts:1]: add this to tg_edits.md

File: fulp_modules/features/antagonists/infiltrator/infil_ruleset.dm [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+		if(!isturf(carp_spawn.loc))
+			stack_trace("Carp spawn found not on a turf: [carp_spawn.type] on [isnull(carp_spawn.loc) ? "null" : carp_spawn.loc.type]")
+			continue
+		spawn_locs += carp_spawn.loc
+	if(!spawn_locs.len)
+		message_admins("No valid spawn locations found, aborting...")
+		return MAP_ERROR
+
+
+	var/list/candidates = get_candidates(ROLE_INFILTRATOR, ROLE_INFILTRATOR)
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrator/infil_ruleset.dm:55]: I think it's supposed to be
```suggestion
	var/list/candidates = get_candidates(ROLE_INFILTRATOR, ROLE_SYNDICATE)
```

File: fulp_modules/features/antagonists/infiltrator/infil_ruleset.dm [hunk 9]
```diff
@@ -0,0 +1,84 @@
+/datum/dynamic_ruleset/midround/from_ghosts/infiltrator
+	name = "Infiltrator"
+	antag_datum = /datum/antagonist/traitor/infiltrator
+	antag_flag = ROLE_INFILTRATOR
+	enemy_roles = list(
+		JOB_CAPTAIN,
+		JOB_DETECTIVE,
+		JOB_HEAD_OF_SECURITY,
+		JOB_SECURITY_OFFICER,
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrator/infil_ruleset.dm:9]: Detective, but not Warden?

File: fulp_modules/_global_vars.dm [hunk 10]
```diff
@@ -56,6 +56,7 @@ GLOBAL_LIST_INIT(fulp_ban_list, list(
 	"Antagonist Positions" = list(
 		ROLE_BLOODSUCKER,
 		ROLE_MONSTERHUNTER,
+		ROLE_INFILTRATOR
```
Comment (JohnFulpWillard) [inline fulp_modules/_global_vars.dm:59]:
```suggestion
		ROLE_INFILTRATOR,
```

File: code/__DEFINES/role_preferences.dm [hunk 11]
```diff
@@ -154,6 +154,7 @@ GLOBAL_LIST_INIT(special_roles, list(
 	ROLE_VAMPIRICACCIDENT = 0,
 	ROLE_BLOODSUCKERBREAKOUT = 0,
 	ROLE_MONSTERHUNTER = 0,
+	ROLE_INFILTRATOR = 14
```
Comment (JohnFulpWillard) [inline code/__DEFINES/role_preferences.dm:157]:
```suggestion
	ROLE_INFILTRATOR = 0,
```

File: fulp_modules/features/antagonists/infiltrator/infiltrators.dm [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+
+/datum/antagonist/traitor/infiltrator/on_gain()
+	if(give_objectives)
+		forge_traitor_objectives()
+	if(give_equipment)
+		equip_infiltrator(owner.current)
+	owner.current.mind.set_assigned_role(SSjob.GetJobType(/datum/job/infiltrator))
+	owner.current.mind.special_role = ROLE_INFILTRATOR
+	infil_flavor = strings("infiltrator_flavor.json", "infiltrator", "fulp_modules/strings/infils")
+	..()
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrator/infiltrators.dm:28]:
```suggestion
	return ..()
```

File: fulp_modules/features/antagonists/infiltrator/infiltrators.dm [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+	data["allies"] = infil_flavor["allies"]
+	data["goal"] = infil_flavor["goal"]
+	data["has_uplink"] = uplink ? TRUE : FALSE
+	if(uplink)
+		data["uplink_intro"] = infil_flavor["uplink"]
+		data["uplink_unlock_info"] = uplink.unlock_text
+	data["objectives"] = get_objectives()
+	return data
+
+/datum/outfit/infiltrator
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrator/infiltrators.dm:54]: This outfit is what will be shown on the antag prefs menu btw, if that's your intent then ok 👍

**Review by JohnFulpWillard — COMMENTED** [review]
idk how many times ill have to review but i think this covers most of it

File: fulp_modules/features/antagonists/infiltrator/infiltrators.dm (hunk 6 above)
Comment (SmoSmoSmoSmok) [reply fulp_modules/features/antagonists/infiltrator/infiltrators.dm:74]: Coz like they're meant to be this overly secret existence. WHEN they die all traces of them are wiped its very cool

File: fulp_modules/features/antagonists/infiltrator/infiltrators.dm (hunk 13 above)
Comment (SmoSmoSmoSmok) [reply fulp_modules/features/antagonists/infiltrator/infiltrators.dm:54]: Yeah yeah THEY look badass in the pref menu its like my greatest creation ever in existence

File: fulp_modules/features/antagonists/infiltrator/infiltrators.dm [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+	return infiltrator.equipOutfit(/datum/outfit/infiltrator)
+
+/datum/antagonist/traitor/infiltrator/on_gain()
+	if(give_objectives)
+		forge_traitor_objectives()
+	if(give_equipment)
+		equip_infiltrator(owner.current)
+	owner.current.mind.set_assigned_role(SSjob.GetJobType(/datum/job/infiltrator))
+	owner.current.mind.special_role = ROLE_INFILTRATOR
+	infil_flavor = strings("infiltrator_flavor.json", "infiltrator", "fulp_modules/strings/infils")
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrator/infiltrators.dm:27]:
```suggestion
	traitor_flavor = strings("infiltrator_flavor.json", "infiltrator", "fulp_modules/strings/infils")
```

File: fulp_modules/features/antagonists/infiltrator/infiltrators.dm [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+		data["responses"] = jointext(GLOB.syndicate_code_response, ", ")
+	data["theme"] = infil_flavor["ui_theme"]
+	data["code"] = uplink?.unlock_code
+	data["failsafe_code"] = uplink?.failsafe_code
+	data["intro"] = infil_flavor["introduction"]
+	data["allies"] = infil_flavor["allies"]
+	data["goal"] = infil_flavor["goal"]
+	data["has_uplink"] = uplink ? TRUE : FALSE
+	if(uplink)
+		data["uplink_intro"] = infil_flavor["uplink"]
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrator/infiltrators.dm:49]:
```suggestion
	data["theme"] = traitor_flavor["ui_theme"]
	data["code"] = uplink?.unlock_code
	data["failsafe_code"] = uplink?.failsafe_code
	data["intro"] = traitor_flavor["introduction"]
	data["allies"] = traitor_flavor["allies"]
	data["goal"] = traitor_flavor["goal"]
	data["has_uplink"] = uplink ? TRUE : FALSE
	if(uplink)
		data["uplink_intro"] = traitor_flavor["uplink"]
```

File: fulp_modules/features/antagonists/infiltrator/infiltrators.dm [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+	antagpanel_category = "Infiltrator"
+	job_rank = ROLE_INFILTRATOR
+	antag_hud_name = "traitor"
+	hijack_speed = 1
+	show_name_in_check_antagonists = TRUE
+	show_to_ghosts = TRUE
+	antag_moodlet = /datum/mood_event/focused
+	preview_outfit = /datum/outfit/infiltrator
+	var/give_equipment = TRUE
+	var/list/infil_flavor
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrator/infiltrators.dm:12]:
```suggestion
```

**Review by JohnFulpWillard — COMMENTED** [review]
remove infil_flavor, use traitor_flavor

File: fulp_modules/features/antagonists/infiltrator/infiltrators.dm [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+
+/datum/antagonist/traitor/infiltrator/on_gain()
+	if(give_objectives)
+		forge_traitor_objectives()
+	if(give_equipment)
+		equip_infiltrator(owner.current)
+	owner.current.mind.set_assigned_role(SSjob.GetJobType(/datum/job/infiltrator))
+	owner.current.mind.special_role = ROLE_INFILTRATOR
+	infil_flavor = strings("infiltrator_flavor.json", "infiltrator", "fulp_modules/strings/infils")
+	return ..()
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrator/infiltrators.dm:28]:
```suggestion
/datum/antagonist/traitor/infiltrator/on_gain()
	. = ..()
	if(give_equipment)
		equip_infiltrator(owner.current)
	owner.current.mind.set_assigned_role(SSjob.GetJobType(/datum/job/infiltrator))
	owner.current.mind.special_role = ROLE_INFILTRATOR
	infil_flavor = strings("infiltrator_flavor.json", "infiltrator", "fulp_modules/strings/infils")
```

**Comment by Pepsilawn** [comment]
old infils was more unique than this :-(

File: fulp_modules/features/antagonists/infiltrator/infiltrators.dm [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+	glasses = /obj/item/clothing/glasses/night
+	mask = /obj/item/clothing/mask/gas/syndicate
+	head = /obj/item/clothing/head/helmet/space/syndicate
+	ears = /obj/item/radio/headset
+	back = /obj/item/storage/backpack/old
+	backpack_contents = list(
+		/obj/item/storage/box/survival/syndie = 1,
+		/obj/item/knife/combat/survival = 1,
+	)
+	r_hand = /obj/item/card/id/advanced/chameleon
```
Comment (SgtHunk) [inline fulp_modules/features/antagonists/infiltrator/infiltrators.dm:40]: why not on their id slot

File: fulp_modules/features/antagonists/infiltrator/infiltrators.dm [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+	l_hand = /obj/item/tank/jetpack
+	shoes = /obj/item/clothing/shoes/jackboots
+	gloves = /obj/item/clothing/gloves/combat
+	r_pocket = /obj/item/tank/internals/emergency_oxygen
+	internals_slot = ITEM_SLOT_RPOCKET
+	belt = /obj/item/crowbar/power/syndicate
```
Comment (SgtHunk) [inline fulp_modules/features/antagonists/infiltrator/infiltrators.dm:46]: giving them the 2nd best crowbar in the game 😟

**Review by SgtHunk — COMMENTED** [review]
watered down ☹️

**Comment by SmoSmoSmoSmok** [comment]
some nake d chillean man broke into my home and beat me up im going to go chase him

**Pull request closed without merging**